legal-api/src/legal_api/core/filing.py

```diff
@@ -90,17 +115,35 @@ def payment_account(self, account_identifier):
         self._payment_account = account_identifier
         self.storage.payment_account = self._payment_account
 
+    @property
+    def status(self) -> str:
+        """Return the status of this Filing."""
+        if not self._status and self._storage:
+            self._status = self._storage.status
+        return self._status
+
     @property
     def json(self) -> Optional[Dict]:
         """Return a dict representing the filing json."""
         _json = {}
         if self._storage:
             if self._storage.status == Filing.Status.COMPLETED.value:
                 _json = VersionedBusinessDetailsService.get_revision(self.id, self._storage.business_id)
+
+                if self.filing_type == Filing.FILING_TYPES.CORRECTION.value:
```

Comment:
Should this be outside the completed check? We have the reports available even before the filing gets completed and that time also we may need this diff

Reply:
These are all the states:
```
        COMPLETED = 'COMPLETED'
        DRAFT = 'DRAFT'
        EPOCH = 'EPOCH'
        ERROR = 'ERROR'
        PAID = 'PAID'
        PENDING = 'PENDING'
        PAPER_ONLY = 'PAPER_ONLY'
        PENDING_CORRECTION = 'PENDING_CORRECTION'
```
So probably should be available only in 
```
        COMPLETED = 'COMPLETED'
        PAID = 'PAID'
        PENDING = 'PENDING'
```

Reply:
Yes, I think so too

Reply:
Cool, on it. thx